### PR TITLE
"Week-9 Solutions" 

### PR DESCRIPTION
**Pull request by josephfox-ch**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
+//! Solution to 1st problem
+
+// This function edits the incoming text to capitalize the first letter of the words.
+function capitalizeFirstLetter(str) {
+  let capitalStr = "";
+  for (let index = 0; index < str.length; index++) {
+    let currentChar = str[index];
+    let previousChar = str[index - 1];
+    // If this character is at the beginning of the word or the previous character is a space, convert to uppercase.
+    if (index == 0 || previousChar == " ") {
```
Comment (hi-coders) [inline app.js:10]: you could also use split method!

Comment (hi-coders) [inline app.js:10]: I am not sure about the effciency of this algorithm. You have to visit all chars! split gives you the words! It is more efficient

File: app.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+// This function takes an array of full names and extracts and capitalizes the initials of each name.
+const namesArr = [
+  "roronoa zoro",
+  "vinsmoke sanji",
+  "tony tony chopper",
+  "gold roger",
+];
+
+function selectAndCapitalizeInitials(arr) {
```
Comment (hi-coders) [inline app.js:33]: select and cpitalize, basically your function is doing two jobs. Or at least the function names tell me that.
Choose a correct name or create  two functions and separate the jobs, if any

File: app.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+selectAndCapitalizeInitials(namesArr);
+
+//! Solution to 3rd problem
+// This function checks if the input string represents a valid date, and if so, it determines if it's a weekend.
+let isWeekendOrNot = (dateInput) => {
+  // Attempt to create a Date object from the input.
+  const userDate = new Date(dateInput);
+  // Check if the userDate is a valid Date object and not NaN.
+  if (userDate instanceof Date && !isNaN(userDate)) {
```
Comment (hi-coders) [inline app.js:58]: good practice

File: app.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    const userDay = userDate.getDay();
+    // Check if it's a weekend (Saturday or Sunday) and display the result.
+    if (userDay == 0 || userDay == 6) {
+      console.log("Today is weekend");
```
Comment (hi-coders) [inline app.js:62]: separate presentation from the business logic

File: app.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+isWeekendOrNot("35/14/2003");
+isWeekendOrNot("15/23/2003");
+isWeekendOrNot("08-09-2003");
+isWeekendOrNot("2003-6-4");
+
+//! Solution to 4th problem
+// This function finds the current day and month and displays them in a human-readable format.
+function findCurrentDayAndMonth() {
+  // Get the current date.
+  const CurrentDate = new Date();
```
Comment (hi-coders) [inline app.js:84]: camel case!

**Review by hi-coders — CHANGES_REQUESTED** [review]

File: app.js (hunk 1 above)
Comment (josephfox-ch) [reply app.js:10]: yes that is right. it would be better I think

Comment (josephfox-ch) [reply app.js:10]: Actually, I wanted to solve the problem as a whole without depending on any single factor. The split() method is of course a nice method that solves the problem, but when I faced a real and complex problem and thought that the sentence could be more complex, I thought that this algorithm would use commas, periods, exclamation points, etc. even if there were no spaces between the words. I wanted it to work despite the factors. However, I have never thought about the possibility of splitting according to more than one alternative in the split method, and I don't know. For example, divide by comma or space or period or exclamation and make an array.

File: app.js (hunk 4 above)
Comment (josephfox-ch) [reply app.js:62]: I researched this principle and will pay more attention from now on. Thank you very much for your contribution.

Comment (josephfox-ch) [reply app.js:62]: I researched this principle and will pay more attention from now on. Thank you very much for your contribution.

File: app.js (hunk 5 above)
Comment (josephfox-ch) [reply app.js:84]: sorry it escaped my attention

**Pull request merged**